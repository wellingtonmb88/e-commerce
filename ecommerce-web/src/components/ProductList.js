import React, { Component } from "react";
import PropTypes from "prop-types";
import { List, ListItem, ListItemText } from "@material-ui/core";

const styles = {
  list: { margin: "10px 50px 10px 50px" }
};

class ProductList extends Component {
  generateListItem = item => {
    const { onItemClick } = this.props;
    return (
      <div key={item.id}>
        <ListItem
          divider={true}
        >
          <ListItemText primary={item.title} />
          <ListItemText primary={item.price} />
          <ListItemText primary={item.description} />
        </ListItem>
      </div>
    );
  };

  render() {
    const { items } = this.props;
    return (
      <List component="nav" style={styles.list}>
        {items.map(this.generateListItem)}
      </List>
    );
  }
}

ProductList.propTypes = {
  items: PropTypes.array.isRequired
};

export default ProductList;
