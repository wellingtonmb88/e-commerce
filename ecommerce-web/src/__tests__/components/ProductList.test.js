import React from "react";
import { shallow } from "enzyme";
import ProductList from "components/ProductList";

const props = {
  items: [
    {
      id: 1,
      title: "Ferrari 1984",
      price: 3000.99,
      description: "Ferari 1984, Red, 1000hp "
    },
    {
      id: 2,
      title: "Porsche 2015 ",
      price: 6000.99,
      description: "Porsche 2015, Black, 2000hp "
    }
  ]
};

test("renders ProductList without crashing", () => {
  const tree = shallow(<ProductList {...props} />);
  expect(tree).toMatchSnapshot();
});
