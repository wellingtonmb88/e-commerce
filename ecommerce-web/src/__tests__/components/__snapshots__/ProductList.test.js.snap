// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductList without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    items={
        Array [
            Object {
              "description": "Ferari 1984, Red, 1000hp ",
              "id": 1,
              "price": 3000.99,
              "title": "Ferrari 1984",
            },
            Object {
              "description": "Porsche 2015, Black, 2000hp ",
              "id": 2,
              "price": 6000.99,
              "title": "Porsche 2015 ",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <WithStyles(ListItem)
                    divider={true}
          >
                    <WithStyles(ListItemText)
                              primary="Ferrari 1984"
                    />
                    <WithStyles(ListItemText)
                              primary={3000.99}
                    />
                    <WithStyles(ListItemText)
                              primary="Ferari 1984, Red, 1000hp "
                    />
          </WithStyles(ListItem)>
</div>,
        <div>
          <WithStyles(ListItem)
                    divider={true}
          >
                    <WithStyles(ListItemText)
                              primary="Porsche 2015 "
                    />
                    <WithStyles(ListItemText)
                              primary={6000.99}
                    />
                    <WithStyles(ListItemText)
                              primary="Porsche 2015, Black, 2000hp "
                    />
          </WithStyles(ListItem)>
</div>,
      ],
      "component": "nav",
      "style": Object {
        "margin": "10px 50px 10px 50px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(ListItem)
            divider={true}
>
            <WithStyles(ListItemText)
                        primary="Ferrari 1984"
            />
            <WithStyles(ListItemText)
                        primary={3000.99}
            />
            <WithStyles(ListItemText)
                        primary="Ferari 1984, Red, 1000hp "
            />
</WithStyles(ListItem)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItemText)
                primary="Ferrari 1984"
/>,
              <WithStyles(ListItemText)
                primary={3000.99}
/>,
              <WithStyles(ListItemText)
                primary="Ferari 1984, Red, 1000hp "
/>,
            ],
            "divider": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "Ferrari 1984",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": 3000.99,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "Ferari 1984, Red, 1000hp ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(ListItem)
            divider={true}
>
            <WithStyles(ListItemText)
                        primary="Porsche 2015 "
            />
            <WithStyles(ListItemText)
                        primary={6000.99}
            />
            <WithStyles(ListItemText)
                        primary="Porsche 2015, Black, 2000hp "
            />
</WithStyles(ListItem)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItemText)
                primary="Porsche 2015 "
/>,
              <WithStyles(ListItemText)
                primary={6000.99}
/>,
              <WithStyles(ListItemText)
                primary="Porsche 2015, Black, 2000hp "
/>,
            ],
            "divider": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "Porsche 2015 ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": 6000.99,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "primary": "Porsche 2015, Black, 2000hp ",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(ListItem)
                        divider={true}
            >
                        <WithStyles(ListItemText)
                                    primary="Ferrari 1984"
                        />
                        <WithStyles(ListItemText)
                                    primary={3000.99}
                        />
                        <WithStyles(ListItemText)
                                    primary="Ferari 1984, Red, 1000hp "
                        />
            </WithStyles(ListItem)>
</div>,
          <div>
            <WithStyles(ListItem)
                        divider={true}
            >
                        <WithStyles(ListItemText)
                                    primary="Porsche 2015 "
                        />
                        <WithStyles(ListItemText)
                                    primary={6000.99}
                        />
                        <WithStyles(ListItemText)
                                    primary="Porsche 2015, Black, 2000hp "
                        />
            </WithStyles(ListItem)>
</div>,
        ],
        "component": "nav",
        "style": Object {
          "margin": "10px 50px 10px 50px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(ListItem)
              divider={true}
>
              <WithStyles(ListItemText)
                            primary="Ferrari 1984"
              />
              <WithStyles(ListItemText)
                            primary={3000.99}
              />
              <WithStyles(ListItemText)
                            primary="Ferari 1984, Red, 1000hp "
              />
</WithStyles(ListItem)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItemText)
                  primary="Ferrari 1984"
/>,
                <WithStyles(ListItemText)
                  primary={3000.99}
/>,
                <WithStyles(ListItemText)
                  primary="Ferari 1984, Red, 1000hp "
/>,
              ],
              "divider": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Ferrari 1984",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": 3000.99,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Ferari 1984, Red, 1000hp ",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(ListItem)
              divider={true}
>
              <WithStyles(ListItemText)
                            primary="Porsche 2015 "
              />
              <WithStyles(ListItemText)
                            primary={6000.99}
              />
              <WithStyles(ListItemText)
                            primary="Porsche 2015, Black, 2000hp "
              />
</WithStyles(ListItem)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItemText)
                  primary="Porsche 2015 "
/>,
                <WithStyles(ListItemText)
                  primary={6000.99}
/>,
                <WithStyles(ListItemText)
                  primary="Porsche 2015, Black, 2000hp "
/>,
              ],
              "divider": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Porsche 2015 ",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": 6000.99,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Porsche 2015, Black, 2000hp ",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
